instrumentation/action_mailer/lib/opentelemetry/instrumentation/action_mailer/instrumentation.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActionMailer
+      # The Instrumentation class contains logic to detect and install the ActionMailer instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('6.1.0')
+        install do |_config|
+          resolve_email_address
+          ecs_mail_convention
+          require_dependencies
+        end
+
+        present do
+          defined?(::ActionMailer)
+        end
+
+        compatible do
+          gem_version >= MINIMUM_VERSION
+        end
+
+        option :disallowed_notification_payload_keys, default: [], validate: :array
+        option :notification_payload_transform,       default: nil, validate: :callable
+        option :email_address,                        default: :omit, validate: %I[omit include]
+
+        private
+
+        def gem_version
+          ::ActionMailer.version
+        end
+
+        def resolve_email_address
+          return unless ActionMailer::Instrumentation.instance.config[:email_address] == :omit
+
+          ActionMailer::Instrumentation.instance.config[:disallowed_notification_payload_keys] += ['email.to.address', 'email.from.address', 'email.cc.address', 'email.bcc.address']
+        end
+
+        def ecs_mail_convention
+          if ActionMailer::Instrumentation.instance.config[:notification_payload_transform].nil?
+            transform_attributes = lambda do |payload|
+              transform_payload(payload)
+            end
+          else
+            original_callable = ActionMailer::Instrumentation.instance.config[:notification_payload_transform]
+            transform_attributes = lambda do |payload|
+              original_callable.call(payload)
+              transform_payload(payload)
+            end
+          end
+          ActionMailer::Instrumentation.instance.config[:notification_payload_transform] = transform_attributes
+        end
+
+        # email attribute key convention is obtained from: https://www.elastic.co/guide/en/ecs/8.11/ecs-email.html
+        def transform_payload(payload)
+          payload['email.message_id'] = payload[:message_id]
+          payload['email.subject']    = payload[:subject]
+          payload['email.x_mailer']   = payload[:mailer]
+          payload['email.to.address'] = payload[:to]
+          payload['email.from.address'] = payload[:from]
+          payload['email.cc.address'] = payload[:cc]
+          payload['email.bcc.address'] = payload[:bcc]
+          payload['email.delivery_timestamp'] = payload[:date]
+          payload['email.origination_timestamp'] = payload[:date]
+
+          payload.delete_if { |item| item.instance_of?(Symbol) }
```

Comment:
Why delete symbols? Would making them strings satisfy the API?

Reply:
If not removing the symbol, they will show in span attributes, which they contain the same value as `email.***` attributes. 

active_support notification expose all key/value from payload unless there is key in `disallowed_notification_payload_keys`, but that will have to include a lot of keys; so I just remove them here.

Reply:
I see. Do you think it will be too expensive to allocate a new hash here?

Reply:
You mean `delete_if`? I don't think `delete_if` create new hash but correct me if I am wrong.

Reply:
The `transform_payload` should return a new hash.

The problem I think is that this is mutating the payload, which will impact other downstream subscribers that will be expecting those fields:

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/8336decf0e28712c98bb010744e677c0d02f6f95/instrumentation/active_support/lib/opentelemetry/instrumentation/active_support/span_subscriber.rb#L81

Reply:
Updated!